vignettes/EpiNow2.Rmd

```diff
@@ -87,6 +87,20 @@ example_incubation_period
 #>         0.069
 ```
 
+Users can also pass a non-parametric delay distribution vector using the `NonParametric` option 
+for both the generation interval and reporting delays. It is important to note that if doing so,
+both delay distributions are 0-indexed, meaning the first element corresponds to the probability mass
+at day 0 of an individual's infection. Because the discretised renewal equation doesn't support mass on day 0, the generation interval should be passed in as a 0-indexed vector with a mass of zero on day 0. 
+If this is not the case, a warning will indicate that the vector is being left-truncated and renormalized.
+
+```r
+example_non_parametric_gi <-  NonParametric(pmf = c(0, 0.3, 0.5, 0.2))
```

Comment:
I know this has been merged but given the preceding text, I would have expected to see the warning, i.e., the example would use `gt_opts()` like so

```r
> gt_opts(NonParametric(pmf = c(0.1, 0.3, 0.5, 0.2)))
- nonparametric distribution
  PMF: [0.091 0.27 0.45 0.18]
Warning message:
Specifying nonparametric generation times with nonzero first element was deprecated in
EpiNow2 1.6.0.
ℹ Since zero generation times are not supported by the model, the generation time will be
  left-truncated at one.
ℹ In future versions this will cause an error. Please ensure that the first element of
  the nonparametric generation interval is zero.
ℹ The deprecated feature was likely used in the EpiNow2 package.
  Please report the issue at <https://github.com/epiforecasts/EpiNow2/issues>.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated. 
```

Reply:
@jamesmbaazam Do you think it makes sense to include a non-parametric pmf that produces this warning as you have shown?

The one currently included has a value of 0 on day 0, so it won't produce the warning! Perhaps the preceding text isn't clear enough.

Reply:
I think the text is fine but personally would have expected the code sample to showcase the warning thrown when the delay is not 0-indexed. Additionally, I would suggest to explicitly print the results of  `example_non_parametric_gi` and `example_non_parametric_delay` in the vignette.

Reply:
I can open an issue and subsequent PR to add this.

Do you think there should be an example for both the correct specification (so with the 0 on day 0) and the incorrect specification that results in a warning (as you demonstrated)?  My only concern is about bloating the vignette, otherwise I think it could make sense to show both.

Reply:
I'm not convinced we want to show a way of specifying that we're discouraging in the warning (and that will cause an error in the future). Perhaps we should just remove the highlighted sentence if it's confusing?

Reply:
@sbfnk You're right about not encouraging the wrong specification. 

Maybe, we can reword this part "If this is not the case, a warning will indicate that the vector is being left-truncated and normalized." -> "If this is not the case, the vector will be left-truncated and normalized."

I do see that the doc of `gt_opts()` has the following wording: "Because the discretised renewal equation used in the package does not support zero generation times, any distribution specified here will be left-truncated at one, i.e. the first element of the nonparametric or discretised probability distribution used for the generation time is set to zero and the resulting distribution renormalised." Shouldn't we just reuse that here?

Reply:
I still think that in the vignette we should only tell users how to specify this correctly and not what happens if they fail to do so (which they'll find out with the warning anyway). So I'd vote for removing the sentence altogether.

Reply:
@kaitejohnson Would you like to take this one 😃 ?